File: GameRanks/src/main/java/GameRanks/GameRanks/clientStruct/page/UserPageStruct.java
package GameRanks.GameRanks.clientStruct.page;

import GameRanks.GameRanks.clientStruct.element.UserPageReviewStruct;
import GameRanks.GameRanks.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UserPageStruct {
    private User user;
    private Iterable<UserPageReviewStruct> userPageReviewStructs;
}

File: GameRanksClient/src/app/models/serverStructs/page/UserPageStruct.ts
import {User} from "../../entity/User";
import {UserPageReviewStruct} from "../../entity/UserPageReviewStruct";

export class UserPageStruct{
	user: User;
	userPageReviewStructs: UserPageReviewStruct[];

	constructor(user: User, userPageReviewStructs: UserPageReviewStruct[]){
		this.user = user;
		this.userPageReviewStructs = userPageReviewStructs;
	}
}
